packages/dds/merge-tree/src/localReference.ts

```diff
@@ -154,8 +154,9 @@ class LocalReference implements LocalReferencePosition {
  */
 export function createDetachedLocalReferencePosition(
 	refType?: ReferenceType,
+	slidingPreference?: SlidingPreference,
```

Comment:
```suggestion
	slidingPreference: SlidingPreference | undefined,
```

Forcing the sliding preference to be passed, even if undefined can help prevent bugs like this, as it makes it a little harder to miss, unlike optional params which are explicitly able to be missed.

Reply:
For sliding preference, it probably makes sense that at some layer, like the user facing api, has this as optional. it might be worth it to look at the stack, and see if it optional, but always passed, as those are good places to consider making it non-optional, but able to be undefined.